server/feeders/reviewsgenerator.js

```diff
@@ -36,133 +38,259 @@ var fiwareHeaders = {
-      'attributes': attr
-    }, returnPost);
-  });
-};
+  if (typeof argv.restaurant === 'string' &&
+      argv.restaurant !== '') {
+    restaurantSelected = argv.restaurant;
+  }
 
-var triggerRestaurantsRatings = function() {
+  if (typeof argv.numberOfReviews === 'number' &&
+      argv.numberOfReviews !== '') {
+    numberOfReviews = argv.numberOfReviews;
+  }
 
-  var returnPost = function(data) {
-    restaurantsModified++;
-    console.log(restaurantsModified + '/' + restaurantsData.length);
-  };
+  if (argv.organization && argv.restaurant) {
+    console.error('\'organization\' and \'restaurant\' ' +
+                  'are incompatible. Please try again.');
+    showHelp();
+    process.exit(1);
+  }
 
-  var q = async.queue(function(task, callback) {
-    var restaurantId = utils.generateId(task.restaurantName);
-    setTimeout(function() {
-      utils.getListByType(RESTAURANT_TYPE, restaurantId, fiwareHeaders)
-      .then(function(data) {
-        var fwHeaders = utils.completeHeaders(fiwareHeaders,
-                                              data.body.department);
-        return utils.sendRequest('PATCH', task.aggregateRatings, restaurantId,
-                                 fwHeaders);
-      })
-      .then(callback)
-      .catch(function(err) {
-        console.log(err.error);
-      });
-    }, delay);
-
-  }, apiRestSimtasks);
-
-  q.drain = function() {
-    console.log('Total restaurants modified: ' + restaurantsModified);
-  };
+  if (!argv.restaurant && argv.numberOfReviews) {
+    console.error('\'numberOfReviews\' needs a \'restaurant\' ' +
+                  'to load into. Please try again.');
+    showHelp();
+    process.exit(1);
+  }
+}
 
-  restaurantsData.forEach(function(element) {
-    var restaurantName = element.name;
+/**
+ * Function to load the restaurant information into a variable
+ *
+ * @param {Object} data - Restaurants information
+*/
+function processRestaurants(data) {
+  restaurantsData = utils.objectToArray(JSON.parse(JSON.stringify(data.body)));
+  feedOrionReviews();
+}
+
+/**
+ * Function to get the restaurant information for the reviews
+*/
+function getRestaurantInformation() {
+  if (organization) {
     var filter = [];
     var queryString = {};
-    utils.addConditionToQuery(filter, 'itemReviewed', '==', restaurantName);
+    utils.addConditionToQuery(filter, 'department', '==', organization);
     queryString.q = filter.join(';');
-    utils.getListByType(REVIEW_TYPE, null, fiwareHeaders, queryString)
-    .then(function(restaurantReviews) {
-      var ratings = utils.getAggregateRating(restaurantReviews.body);
-      q.push({'aggregateRatings': ratings, 'restaurantName': restaurantName},
-             returnPost);
-    })
+    utils.getListByType(RESTAURANT_TYPE, null, fiwareHeaders, queryString)
+    .then(processRestaurants)
     .catch(function(err) {
-      console.log(err.error);
+      console.error(err.error);
     });
-  });
-};
+  }  else if (restaurantSelected) {
```

Comment:
We referred that could be clearer to put all the if block inside curly braces (`{` `}`) instead of use an *elif* like style.

You have:

```
if (organization) {
  ...
}
else if (restaurantSelected) {
  ...
} else {
  ...
}
```

and we suggest:

```
if (organization) {
  ...
}
else {
  if (restaurantSelected) {
    ...
  } else {
    ...
  }
}
```

This is only a code style suggestion, not an error. :)